desc/backend.py

```diff
@@ -744,3 +753,37 @@ def root(
         """
         out = scipy.optimize.root(fun, x0, args, jac=jac, tol=tol)
         return out.x, out
+
+    def take(
+        a,
+        indices,
+        axis=None,
+        out=None,
+        mode="fill",
+        unique_indices=False,
+        indices_are_sorted=False,
+        fill_value=None,
+    ):
+        """A numpy implementation of jnp.take."""
```

Comment:
If the user has `jax` installed, then the function `take` is an alias for `jnp.take`. If the user only has numpy installed, then `take` is aliased to this function which is just `np.take` generalized to support the API of `jnp.take`. In particular `jnp.take` supports additional arguments `unique_indices` and `indices_are_sorted` to execute more efficiently on certain hardware and an additional `mode="fill"` option which is often necessary in jitable algorithms due to JAX's dynamic size array limitations.

If we didn't do this, DESC would error for users without JAX installed. i think the current policy is to keep it compatible with numpy in case a jax is abandoned for a new ad library